pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -103,54 +212,253 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
+	// deploy machine config pool
+	mcp := machineconfigpool.NewPerformance(profile)
+	if err := controllerutil.SetControllerReference(profile, mcp, r.scheme); err != nil {
+		return err
 	}
-	return &corev1.Pod{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      cr.Name + "-pod",
-			Namespace: cr.Namespace,
-			Labels:    labels,
-		},
-		Spec: corev1.PodSpec{
-			Containers: []corev1.Container{
-				{
-					Name:    "busybox",
-					Image:   "busybox",
-					Command: []string{"sleep", "3600"},
-				},
-			},
-		},
+	if err := r.createOrUpdateMachineConfigPool(mcp); err != nil {
+		return err
+	}
+
+	// pause machine config pool
+	if err := r.pauseMachineConfigPool(mcp.Name, true); err != nil {
+		return err
+	}
+
+	// deploy machine config
+	mc, err := machineconfig.NewPerformance(r.assetsDir, profile)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profile, mc, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateMachineConfig(mc); err != nil {
+		return err
+	}
+
+	// deploy kubelet config
+	kc := kubeletconfig.NewPerformance(profile)
+	if err := controllerutil.SetControllerReference(profile, kc, r.scheme); err != nil {
+		return err
+	}
+	if err := r.createOrUpdateKubeletConfig(kc); err != nil {
+		return err
+	}
+
+	// deploy feature gate
+	fg := featuregate.NewLatencySensitive()
+	// TOOD: uncomment once https://bugzilla.redhat.com/show_bug.cgi?id=1788061 fixed
+	// if err := controllerutil.SetControllerReference(profile, fg, r.scheme); err != nil {
+	// 	return err
+	// }
+	if err := r.createOrUpdateFeatureGate(fg); err != nil {
+		return err
+	}
+
+	// deploy network latency tuned
+	networkLatencyTuned, err := tuned.NewNetworkLatency(r.assetsDir)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profile, networkLatencyTuned, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateTuned(networkLatencyTuned); err != nil {
+		return err
+	}
+
+	// deploy real time kernel tuned
+	realTimeKernelTuned, err := tuned.NewWorkerRealTimeKernel(r.assetsDir, profile)
+	if err != nil {
+		return err
+	}
+
+	if err := controllerutil.SetControllerReference(profile, realTimeKernelTuned, r.scheme); err != nil {
+		return err
+	}
+
+	if err := r.createOrUpdateTuned(realTimeKernelTuned); err != nil {
+		return err
+	}
+
+	// unpause machine config pool
+	return r.pauseMachineConfigPool(mcp.Name, false)
+}
+
+func (r *ReconcilePerformanceProfile) pauseMachineConfigPool(mcpName string, pause bool) error {
+	// TODO: in case if we will change nodeSelector to the machineConfigPoolSelector under the API,
+	// we will need to fetch the name of the machine config pool on runtime
+	updatedMcp, err := r.getMachineConfigPool(mcpName)
+	if errors.IsNotFound(err) {
+		klog.Warning("failed to set pause, the machine config pool does not exist, probably it was deleted")
+		return nil
+	}
+
+	if err != nil {
+		return err
+	}
+
+	updatedMcp.Spec.Paused = pause
```

Comment:
only perform the update here if updatedMcp.Spec.Paused does not already match the desired pause value.

Reply:
done